source/extensions/common/aws/credentials_provider_impl.cc

```diff
@@ -249,10 +339,17 @@ Credentials CredentialsProviderChain::getCredentials() {
 
 DefaultCredentialsProviderChain::DefaultCredentialsProviderChain(
     Api::Api& api, const MetadataCredentialsProviderBase::MetadataFetcher& metadata_fetcher,
-    const CredentialsProviderChainFactories& factories) {
+    const CredentialsProviderChainFactories& factories, const bool enable_credentials_file) {
   ENVOY_LOG(debug, "Using environment credentials provider");
   add(factories.createEnvironmentCredentialsProvider());
 
+  if (enable_credentials_file) {
```

Comment:
Is it possible to check `Runtime::runtimeFeatureEnabled("envoy.reloadable_features.enable_aws_credentials_file");` directly here instead of passing the bool flag? Any advantage in passing the flag over?

We can also enable and disable the feature flag while unit testing. The extensions or plugin don't have to know whether it has to enable or disable the feature to read from credentials file. It is handled under the hood.

When we finally remove the feature flag the change will be just in this constructor.

Reply:
No advantage, your way is better. Updated.